src/geo/projection/globe_tile_transform.js

```diff
@@ -0,0 +1,148 @@
+// @flow
+import type Transform from '../transform.js';
+import {CanonicalTileID, UnwrappedTileID} from '../../source/tile_id.js';
+import {mat4, vec4, vec3} from 'gl-matrix';
+import MercatorCoordinate, {mercatorZfromAltitude, mercatorXfromLng, mercatorYfromLat} from '../mercator_coordinate.js';
+import EXTENT from '../../data/extent.js';
+import {degToRad, clamp} from '../../util/util.js';
+import {
+    latLngToECEF,
+    globeTileLatLngCorners,
+    globeTileBounds,
+    calculateGlobeMatrix,
+    denormalizeECEF,
+    NORMALIZATION_BIT_RANGE,
+    GLOBE_RADIUS
+} from './globe.js';
+
+function tileNormalizationScale(id: CanonicalTileID) {
+    const bounds = globeTileBounds(id);
+    const maxExtInv = 1.0 / Math.max(...vec3.sub([], bounds.max, bounds.min));
+    const st = (1 << (NORMALIZATION_BIT_RANGE - 1)) - 1;
+    return st * maxExtInv;
+}
+
+export default class GlobeTileTransform {
+    _tr: Transform;
+    _worldSize: number;
+    _globeMatrix: Float64Array;
+
+    constructor(tr: Transform, worldSize: number) {
+        this._tr = tr;
+        this._worldSize = worldSize;
+        this._globeMatrix = calculateGlobeMatrix(tr, worldSize);
+    }
+
+    createTileMatrix(id: UnwrappedTileID): mat4 {
+        const decode = denormalizeECEF(globeTileBounds(id.canonical));
+        return mat4.multiply([], this._globeMatrix, decode);
+    }
+
+    createInversionMatrix(id: UnwrappedTileID): mat4 {
```

Comment:
Investigate whether we could make this matrix identity when globe isn't transitioning between projections.